lib/aws-record/record/batch.rb

```diff
@@ -0,0 +1,53 @@
+# Copyright 2015-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+module Aws
+  module Record
+    class Batch
+      include Enumerable
+
+      attr_reader :items
+
+      def initialize(model, client, items = [])
+        @model = model
+        @client = client
+        @items = items.map do |item|
+          check_item_class(item)
+        end
+      end
+
+      def each(&_block)
+        return enum_for(:each) unless block_given?
+        @items.each do |item|
+          yield item
+        end
+      end
+
+      def add(item)
+        @items << check_item_class(item) unless @items.include?(item)
```

Comment:
It seems as if you would want a consistent return value here. Either always return the array of items, or maybe an error code if the add isn't successful.